lib/cassandra/protocol/v3.rb

```diff
@@ -190,93 +210,117 @@ def actual_decode(buffer, fields, size, code)
 
         def decode_response(opcode, protocol_version, buffer, size, trace_id)
           case opcode
-          when 0x00 # ERROR
-            code    = buffer.read_int
-            message = buffer.read_string
-
-            case code
-            when 0x1000 then UnavailableErrorResponse.new(nil, nil, code, message, buffer.read_consistency, buffer.read_int, buffer.read_int)
-            when 0x1100 then WriteTimeoutErrorResponse.new(nil, nil, code, message, buffer.read_consistency, buffer.read_int, buffer.read_int, buffer.read_string)
-            when 0x1200 then ReadTimeoutErrorResponse.new(nil, nil, code, message, buffer.read_consistency, buffer.read_int, buffer.read_int, (buffer.read_byte != 0))
-            when 0x2400 then AlreadyExistsErrorResponse.new(nil, nil, code, message, buffer.read_string, buffer.read_string)
-            when 0x2500 then UnpreparedErrorResponse.new(nil, nil, code, message, buffer.read_short_bytes)
-            else
-              ErrorResponse.new(nil, nil, code, message)
-            end
-          when 0x02 # READY
-            READY
-          when 0x03 # AUTHENTICATE
-            AuthenticateResponse.new(buffer.read_string)
-          when 0x06 # SUPPORTED
-            SupportedResponse.new(buffer.read_string_multimap)
-          when 0x08 # RESULT
-            result_type = buffer.read_int
-            case result_type
-            when 0x0001 # Void
-              VoidResultResponse.new(nil, nil, trace_id)
-            when 0x0002 # Rows
-              original_buffer_length = buffer.length
-              column_specs, paging_state = Coder.read_metadata_v3(buffer)
-
-              if column_specs.nil?
-                consumed_bytes  = original_buffer_length - buffer.length
-                remaining_bytes = CqlByteBuffer.new(buffer.read(size - consumed_bytes - 4))
-                RawRowsResultResponse.new(nil, nil, protocol_version, remaining_bytes, paging_state, trace_id)
-              else
-                RowsResultResponse.new(nil, nil, Coder.read_values_v3(buffer, column_specs), column_specs, paging_state, trace_id)
+            when 0x00 # ERROR
```

Comment:
The majority of this method has formatting changes. I added a few else blocks to some cases. Noted below.